export_trt.py

```diff
@@ -1,32 +1,255 @@
+            config=CreateConfig(fp16=fp16,
+                refittable=enable_refit,
+                profiles=[p],
+                load_timing_cache=timing_cache,
+                **config_kwargs
+            ),
+            save_timing_cache=timing_cache
+        )
+        save_engine(engine, path=self.engine_path)
+
+def get_unet_trt_profile(cond_dim, min_bs, opt_bs ,max_bs, min_token_count, opt_token_count, max_token_count, min_width, opt_width, max_width, min_height, opt_height, max_height):
+    profile = {
+            'x' :          ((min_bs * 2, 4, min_height // 8, min_width // 8), (opt_bs * 2, 4, opt_width // 8, opt_height // 8), (max_bs * 2, 4, max_height // 8, max_width // 8)),
+            'timesteps' :  ((min_bs * 2,), (opt_bs * 2,), (max_bs * 2,)),
+            'context' :    ((min_bs * 2, min_token_count // 75 * 77, cond_dim), (opt_bs * 2,  opt_token_count // 75 * 77, cond_dim), (max_bs * 2, max_token_count // 75 * 77, cond_dim)),
+            }
+    return profile
+
+def get_trt_profile_filename(profile):
+    return "unet_w[{},{},{}],h[{},{},{}],b[{},{},{}],t[{},{},{}]" \
+        .format(int(profile['x'][0][3] * 8), int(profile['x'][1][3] * 8),int(profile['x'][2][3] * 8), 
+        int(profile['x'][0][2] * 8), int(profile['x'][1][2] * 8),int(profile['x'][2][2] * 8),
+        int(profile['x'][0][0]/2),int(profile['x'][1][0]/2),int(profile['x'][2][0]/2),
+        int(profile['context'][0][1]/77*75),int(profile['context'][1][1]/77*75),int(profile['context'][2][1]/77*75))
+
+def generate_trt_engine_presets(trt_filename, onnx_filename, profile_512_512_1, profile_512_512_2, profile_512_512_4, profile_768x768x1, profile_768x768x2, profile_768x768x4, use_fp16):
+    
     cond_dim = 768  # XXX should be detected for SD2.0
-    x_min = f"{min_bs * 2}x4x{min_height // 8}x{min_width // 8}"
-    x_max = f"{max_bs * 2}x4x{max_height // 8}x{max_width // 8}"
-    context_min = f"{min_bs * 2}x{min_token_count // 75 * 77}x{cond_dim}"
-    context_max = f"{max_bs * 2}x{max_token_count // 75 * 77}x{cond_dim}"
-    timestamps_min = f"{min_bs * 2}"
-    timestamps_max = f"{max_bs * 2}"
+    profiles = []
+    if profile_512_512_1:
+        profiles.append(get_unet_trt_profile(cond_dim, 1, 1 ,1, 75, 75, 75, 512, 512, 512, 512, 512, 512))
+    if profile_512_512_2:
+        profiles.append(get_unet_trt_profile(cond_dim, 2, 2 ,2, 75, 75, 75, 512, 512, 512, 512, 512, 512))    
+    if profile_512_512_4:
+        profiles.append(get_unet_trt_profile(cond_dim, 4, 4 ,4, 75, 75, 75, 512, 512, 512, 512, 512, 512))
+    if profile_768x768x1:
+        profiles.append(get_unet_trt_profile(cond_dim, 1, 1 ,1, 75, 75, 75, 768, 768, 768, 768, 768, 768)) 
+    if profile_768x768x2:
+        profiles.append(get_unet_trt_profile(cond_dim, 2, 2 ,2, 75, 75, 75, 768, 768, 768, 768, 768, 768))     
+    if profile_768x768x4:
+        profiles.append(get_unet_trt_profile(cond_dim, 4, 4 ,4, 75, 75, 75, 768, 768, 768, 768, 768, 768))
+

```

Comment:
`profile_512_512_1` and `profile_768x768x2` underscore and x is used interchangeable.

Also looks weird that you just repeat all numbers 3 times. Can't we make this more dynamic rather than having bools for all presets ?